src/util.ts

```diff
@@ -10,6 +10,23 @@ export const asyncUnimplemented = (action: string) => async () => {
     throw new Error('Unimplemented on this platform: ' + action);
 };
 
+export const timeoutCondition = async (
+    condition: () => boolean | Promise<boolean>,
+    timeoutInMs = 10000,
+    step = 250
+) => {
+    let elapsedMs = 0;
+
+    while (!(await condition())) {
```

Comment:
If this is a promise and it doesn’t resolve within the timeout, this will just continue to run. I think this is confusingly named, because the timeout is only somewhat predictable for the synchronous case.

Reply:
Sure, but you'll have to give me a better name, then. I can't think of one.

Reply:
`syncTryCondition`?

Reply:
Hm, It returns a `Promise` so sync doesn't make sense actually. How about `retryCondition`?

Also, I would suggest actually keeping track of real elapsed time because this is what I would expect of a timeout.

Reply:
> Also, I would suggest actually keeping track of real elapsed time because this is what I would expect of a timeout.

But why? This is not some generic utility function that we make available to the outside world. And internally, I don't care about the elapsed time, I just want to retry a couple of times.

If you really care about the semantics, we should change `timeout` -> `tries` and `step` -> `pauseBetweenTriesInMs`.

Reply:
Ok, if you change that I am also happy. I know we don’t expose this function, but it had me confused during review.